packages/compass-query-bar/src/components/option-editor.tsx

```diff
@@ -108,7 +112,8 @@ const OptionEditor: React.FunctionComponent<OptionEditorProps> = ({
 }) => {
   const showInsights = usePreference('showInsights', React);
   const editorContainerRef = useRef<HTMLDivElement>(null);
-  const editorRef = useRef<EditorRef>(null);
+  const editorInitialValueRef = useRef<string | false>(false);
```

Comment:
The initial value here is whatever gets passed from the props initially, otherwise you will be tracking edits even if nothing changed when component remounts with existing state. [Here's an reference](https://github.com/mongodb-js/compass/blob/81f0c524608ebac757d66318081615f57c1226f8/packages/compass-aggregations/src/components/stage-editor/stage-editor.tsx#L127-L147) of how this is handled in agg builder editor